### PR TITLE
Moving some things around in Cedar/Thm/Partial

### PR DESCRIPTION
**Pull request by cdisselkoen**
Moving some things around in preparation for the next major lemmas towards PE soundness.

* move `Partial/Evaluation/WellFormed.lean` to `Partial/WellFormed.lean` because the `WellFormed` definitions are used throughout the `Partial` folder, not just in `Partial/Evaluation`.  (And they will soon be used in even more places outside `Partial/Evaluation/`)
* Move theorems about `Partial.evaluateBinaryApp` from `Partial/Evaluation/BinaryApp.lean` to a new dedicated file `Partial/EvaluateBinaryApp.lean`.  The former file now only contains theorems about `Partial.evaluate` applied to `Expr.binaryApp`.  This will allow us to reuse theorems about `Partial.evaluateBinaryApp` in upcoming theorems about `Partial.evaluateValue` without a circular import.
* Similarly for `Partial.evaluateUnaryApp`, `Partial.evaluateGetAttr`, `Partial.evaluateHasAttr`, and `Partial.evaluateCall`
* This PR also includes small tweaks to the proofs being moved around

### CODE REVIEW

File: cedar-lean/Cedar/Thm/Partial/EvaluateCall.lean [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    replace ⟨v, h₃, h₄⟩ := List.mapM_some_implies_all_some h₃ arg h₂
+    cases arg <;> simp only [Option.some.injEq] at h₄
+    subst v ; rename_i v
+    exists v
+  · rename_i h₃
+    replace ⟨arg', h₃, h₄⟩ := List.mapM_none_iff_exists_none.mp h₃
+    cases arg' <;> simp only [Except.ok.injEq] at h₁ h₄
+
+/--
+  something akin to `Partial.EvaluateValue.eval_spec_value`, lifted to lists of `Partial.Value`
```
Comment (khieta) [inline cedar-lean/Cedar/Thm/Partial/EvaluateCall.lean:106]: I don't think there's any `EvaluateValue` file in the repo currently. Coming in a future commit?

**Review by khieta — APPROVED** [review]
Lgtm.

Though it feels a little weird to have `EvaluateBinaryApp.lean` and company in the top-level folder though because I'd expect these to be abstracted away & hidden under `Evaluation/`

File: cedar-lean/Cedar/Thm/Partial/EvaluateCall.lean (hunk 1 above)
Comment (cdisselkoen) [reply cedar-lean/Cedar/Thm/Partial/EvaluateCall.lean:106]: Yep.  Right now the comment should be read as "if we were to have such a thing".

Alternately, I can add the file in this PR with just that theorem (which has a 1-line proof) even though it isn't needed by any of the proofs that exist today on `main`

Comment (khieta) [reply cedar-lean/Cedar/Thm/Partial/EvaluateCall.lean:106]: I think it'd be better to wait to add this lemma until you add the proof that requires it. But fine either way 🙂

**Comment by khieta** [comment]
I think I'd expect a structure like `Evaluation/Evaluate` & `Evaluate/EvaluateValue`, or just a flat structure under `Evaluate`. You've spent more time working with the formalization than me though, so take my suggestions with a grain of salt...

**Comment by cdisselkoen** [comment]
Is this the proposal?
```
Partial/
  Authorization/
  Evaluation/
    Evaluate.lean
    Evaluate/ -- files that are today flat under Evaluation
    EvaluateValue.lean
    EvaluateValue/ -- analogues of Evaluate/ files, as needed
    Lemmas/ -- stuff that needs to be shared by `Evaluate` and `EvaluateValue`
      EvaluateBinaryApp.lean
      EvaluateGetAttr.lean
      ...
```
(Or no `Lemmas/` and have the lemmas files be flat in `Evaluation`

**Comment by khieta** [comment]
I was thinking the lemmas would just be flat in `Evaluation`. But otherwise yes, that's what I was imagining.

**Comment by cdisselkoen** [comment]
Reorg'd to that structure.

**Review by emina — APPROVED** [review]
Looks good!

File: cedar-lean/Cedar/Thm/Partial/Evaluation/Evaluate/GetAttr.lean [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+theorem on_concrete_eqv_concrete_eval {x₁ : Expr} {request : Spec.Request} {entities : Spec.Entities} {attr : Attr} :
+  PartialEvalEquivConcreteEval x₁ request entities →
+  PartialEvalEquivConcreteEval (Expr.getAttr x₁ attr) request entities
+:= by
+  unfold PartialEvalEquivConcreteEval
+  intro ih₁
+  unfold Partial.evaluate Spec.evaluate
+  simp only [ih₁]
+  cases Spec.evaluate x₁ request entities <;> simp only [Except.bind_err, Except.bind_ok]
+  case error e => simp only [Except.map, Except.bind_err]
```
[the remark this answers is not in the document]
Comment (cdisselkoen) [reply cedar-lean/Cedar/Thm/Partial/Evaluation/Evaluate/GetAttr.lean:47]: I'll do a pass for fixing many instances of this at a later time.

File: cedar-lean/Cedar/Thm/Partial/Evaluation/EvaluateUnaryApp.lean [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+-/
+theorem evaluateUnaryApp_wf {pval : Partial.Value} {op : UnaryOp} :
+  Partial.evaluateUnaryApp op pval = .ok pval' → pval'.WellFormed
+:= by
+  unfold Partial.evaluateUnaryApp
+  cases pval <;> simp only [Except.ok.injEq]
+  case residual r => intro h₁ ; subst h₁ ; simp [Partial.Value.WellFormed, Partial.ResidualExpr.WellFormed]
+  case value v =>
+    cases h₁ : Spec.apply₁ op v
+    case error e => simp only [Except.bind_err, false_implies]
```
[the remark this answers is not in the document]
Comment (cdisselkoen) [reply cedar-lean/Cedar/Thm/Partial/Evaluation/EvaluateUnaryApp.lean:74]: I'll do a pass for fixing many instances of this at a later time.

**Pull request merged**